File: longestConsecutiveSequence.java
import java.util.*;
public class Solution {
    public int longestConsecutive(int[] num) {
        // Start typing your Java solution below
        // DO NOT write main() function
        HashSet<Integer> m = new HashSet<Integer>();
        for (int i=0; i < num.length; ++i) {
            m.add(num[i]);
        }
        int longestLength = 0;
        for (int i=0; i < num.length; ++i) {
            int j = num[i];
            int length = 1;
            while (m.contains(j+1)) {
                m.remove(j+1);
                j ++;
                length ++;
            }
            j = num[i];
            while (m.contains(j-1)) {
                m.remove(j-1);
                j --;
                length ++;
            }
            if (length > longestLength) {
                longestLength = length;
            }
        }
        return longestLength;
    }
}

File: longestConsecutiveSequence.cpp
#include "leetcode.h"
class Solution {
public:
    int longestConsecutive(vector<int> &num) {
        unordered_set<int> m;
        for (auto i: num)
            m.insert(i);
        int longestLength = 0;
        for (auto i: num) {
            int length = 1;
            for (int j=i+1; m.find(j) != m.end(); j++) {
                m.erase(j);
                length ++;
            }
            for (int j=i-1; m.find(j) != m.end(); j--) {
                m.erase(j);
                length ++;
            }
            longestLength = max(longestLength, length);
        }
        return longestLength;
    }
};
int main() {
    Solution s;
    vector<int> num;
    num.push_back(100);
    num.push_back(4);
    num.push_back(200);
    num.push_back(1);
    num.push_back(3);
    num.push_back(2);
    cout << s.longestConsecutive(num) << endl;
    return 0;
}
